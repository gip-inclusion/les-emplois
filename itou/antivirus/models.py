from django.db import models
from django.utils.text import capfirst

from itou.files.models import File


class Scan(models.Model):
    file = models.OneToOneField(File, on_delete=models.CASCADE)
    clamav_signature = models.TextField()
    clamav_completed_at = models.DateTimeField(null=True, verbose_name="analyse ClamAV le")
    clamav_infected = models.BooleanField(null=True, verbose_name="fichier infect√© selon ClamAV")
    comment = models.TextField(blank=True, verbose_name="commentaire")

    class Meta:
        verbose_name = "analyse antivirus"
        verbose_name_plural = "analyses antivirus"

    def __str__(self):
        text = f"{capfirst(self._meta.verbose_name)} {self.file_id}"
        if self.clamav_infected:
            text = f"[VIRUS] {text}"
        return text
