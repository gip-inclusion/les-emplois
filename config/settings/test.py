import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
