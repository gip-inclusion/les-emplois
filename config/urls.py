from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
