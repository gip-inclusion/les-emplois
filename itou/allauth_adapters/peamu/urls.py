from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
