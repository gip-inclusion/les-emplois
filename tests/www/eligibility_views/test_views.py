from functools import partial

import pytest
from dateutil.relativedelta import relativedelta
from django.urls import reverse
from django.utils import timezone
from freezegun import freeze_time
from pytest_django.asserts import assertRedirects

from itou.eligibility.models.iae import EligibilityDiagnosis
from tests.approvals.factories import ApprovalFactory
from tests.eligibility.factories import IAEEligibilityDiagnosisFactory
from tests.users.factories import EmployerFactory, JobSeekerFactory, LaborInspectorFactory, PrescriberFactory
from tests.utils.test import parse_response_to_soup, pretty_indented


class TestUpdateEligibilityView:
    @pytest.mark.parametrize(
        "factory,status_code",
        [
            [partial(JobSeekerFactory, for_snapshot=True), 403],
            (partial(PrescriberFactory, membership__organization__authorized=True), 200),
            (partial(PrescriberFactory, membership__organization__authorized=False), 403),
            (PrescriberFactory, 403),
            (partial(EmployerFactory, with_company=True), 403),
            [partial(LaborInspectorFactory, membership=True), 403],
        ],
        ids=[
            "job_seeker",
            "authorized_prescriber",
            "prescriber_with_org",
            "prescriber_no_org",
            "employer",
            "labor_inspector",
        ],
    )
    def test_permissions(self, client, factory, status_code):
        user = factory()
        job_seeker = user if user.is_job_seeker else JobSeekerFactory()
        client.force_login(user)
        url = reverse(
            "eligibility_views:update",
            kwargs={"job_seeker_public_id": job_seeker.public_id},
            query={"back_url": reverse("job_seekers_views:list")},
        )
        response = client.get(url)
        assert response.status_code == status_code

    def test_standalone_no_valid_eligibility_diagnosis(self, client, snapshot):
        prescriber = PrescriberFactory(membership__organization__authorized=True)
        job_seeker = JobSeekerFactory(for_snapshot=True)

        client.force_login(prescriber)

        # Without a eligibility diagnosis
        url = reverse(
            "eligibility_views:update",
            kwargs={"job_seeker_public_id": job_seeker.public_id},
            query={"back_url": reverse("job_seekers_views:list")},
        )
        response = client.get(url)
        assert pretty_indented(parse_response_to_soup(response, "#main")) == snapshot(name="0")

        # With an expired eligibility diagnosis
        IAEEligibilityDiagnosisFactory(job_seeker=job_seeker, from_prescriber=True, expired=True)
        response = client.get(url)
        assert pretty_indented(parse_response_to_soup(response, "#main")) == snapshot(name="0")

        response = client.post(url, {"level_1_1": True})
        assertRedirects(response, reverse("job_seekers_views:list"))

        diag = EligibilityDiagnosis.objects.last_considered_valid(job_seeker=job_seeker, for_siae=None)
        assert diag.is_valid is True
        assert diag.expires_at == timezone.localdate() + relativedelta(months=6)

    @freeze_time("2025-03-22")
    def test_standalone_valid_eligibility_diagnosis(self, client, snapshot):
        prescriber = PrescriberFactory(membership__organization__authorized=True)
        job_seeker = JobSeekerFactory(for_snapshot=True)
        eligibility_diagnosis = IAEEligibilityDiagnosisFactory(
            job_seeker=job_seeker, from_prescriber=True, author_prescriber_organization__for_snapshot=True
        )

        client.force_login(prescriber)
        url = reverse(
            "eligibility_views:update",
            kwargs={"job_seeker_public_id": job_seeker.public_id},
            query={"back_url": reverse("job_seekers_views:list")},
        )
        response = client.get(url)
        assert pretty_indented(parse_response_to_soup(response, "#main")) == snapshot

        # if "shrouded" is present then we don't update the eligibility diagnosis
        response = client.post(url, {"level_1_1": True, "shrouded": "whatever"})
        assertRedirects(response, reverse("job_seekers_views:list"))
        assert [eligibility_diagnosis] == list(
            EligibilityDiagnosis.objects.for_job_seeker_and_siae(job_seeker=eligibility_diagnosis.job_seeker)
        )

        # If "shrouded" is NOT present then we update the eligibility diagnosis
        response = client.post(url, {"level_1_1": True})
        assertRedirects(response, reverse("job_seekers_views:list"))
        new_eligibility_diagnosis = (
            EligibilityDiagnosis.objects.for_job_seeker_and_siae(job_seeker=eligibility_diagnosis.job_seeker)
            .order_by()
            .last()
        )
        assert new_eligibility_diagnosis != eligibility_diagnosis
        assert new_eligibility_diagnosis.author == prescriber

    def test_valid_approval(self, client):
        prescriber = PrescriberFactory(membership__organization__authorized=True)
        job_seeker = JobSeekerFactory(for_snapshot=True)
        ApprovalFactory(user=job_seeker)

        client.force_login(prescriber)
        url = reverse(
            "eligibility_views:update",
            kwargs={"job_seeker_public_id": job_seeker.public_id},
            query={"back_url": reverse("job_seekers_views:list")},
        )
        response = client.get(url)
        assertRedirects(response, reverse("job_seekers_views:list"))
