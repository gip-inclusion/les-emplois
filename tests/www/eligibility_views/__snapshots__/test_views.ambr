# serializer version: 1
# name: TestUpdateEligibilityView.test_standalone_no_valid_eligibility_diagnosis[0]
  '''
  <main class="s-main" id="main" role="main">
      <div aria-atomic="true" aria-live="polite" class="toast-container">
      </div>
      <section class="s-title-02">
          <div class="s-title-02__container container">
              <div class="s-title-02__row row">
                  <div class="s-title-02__col col-12">
                      <div class="c-title">
                          <div class="c-title__main">
                              <h1>
                                  Valider
                  
                  l'éligibilité de Jane DOE
                              </h1>
                          </div>
                      </div>
                  </div>
              </div>
          </div>
      </section>
      <section class="s-section">
          <div class="s-section__container container">
              <div class="s-section__row row">
                  <div class="s-section__col col-12 col-xxl-8 col-xxxl-9">
                      <div class="c-form">
                          <div id="form_errors">
                          </div>
                          <form method="post">
                              <input name="csrfmiddlewaretoken" type="hidden" value="NORMALIZED_CSRF_TOKEN"/>
                              <div class="c-info mb-3">
                                  <button class="c-info__summary">
                                      <span>
                                          Pour valider l'éligibilité IAE du candidat
                                      </span>
                                  </button>
                                  <div class="c-info__detail">
                                      <ul>
                                          <li>
                                              Veuillez vous assurer d’avoir réalisé un diagnostic socio-professionnel dans le cadre d'un entretien individuel.
                  Vous pouvez vous appuyer sur le document
                                              <a aria-label="Diagnostic socio-professionnel des candidats (ouverture dans un nouvel onglet)" class="has-external-link" href="https://communaute.inclusion.gouv.fr/surveys/dsp/create/?proconnect_login=true" rel="noreferrer noopener" target="_blank">
                                                  diagnostic socio-professionnel de référence
                                              </a>
                                              .
                                          </li>
                                          <li>
                                              Il est recommandé de sélectionner le(s) critères(s) administratifs d’éligibilité correspondants.
                                          </li>
                                      </ul>
                                  </div>
                              </div>
                              <div class="">
                                  <fieldset>
                                      <legend class="h3">
                                          Critères administratifs de niveau 1
                                      </legend>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_1_helptext" class="form-check-input" id="id_level_1_1" name="level_1_1" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_1">
                                                  Bénéficiaire du RSA
                                              </label>
                                              <div class="form-text" id="id_level_1_1_helptext">
                                                  Revenu de solidarité active
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_2_helptext" class="form-check-input" id="id_level_1_2" name="level_1_2" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_2">
                                                  Allocataire ASS
                                              </label>
                                              <div class="form-text" id="id_level_1_2_helptext">
                                                  Allocation de solidarité spécifique
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_3_helptext" class="form-check-input" id="id_level_1_3" name="level_1_3" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_3">
                                                  Allocataire AAH
                                              </label>
                                              <div class="form-text" id="id_level_1_3_helptext">
                                                  Allocation aux adultes handicapés
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_4_helptext" class="form-check-input" id="id_level_1_4" name="level_1_4" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_4">
                                                  DETLD (+ 24 mois)
                                              </label>
                                              <div class="form-text" id="id_level_1_4_helptext">
                                                  Demandeur d'emploi de très longue durée (inscrit à France Travail)
                                              </div>
                                          </div>
                                      </div>
                                  </fieldset>
                                  <fieldset>
                                      <legend class="h3">
                                          Critères administratifs de niveau 2
                                      </legend>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_5" name="level_2_5" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_5">
                                                  Niveau d'étude 3 (CAP, BEP) ou infra
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_6" name="level_2_6" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_6">
                                                  Senior (+50 ans)
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_7" name="level_2_7" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_7">
                                                  Jeune (-26 ans)
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_8_helptext" class="form-check-input" id="id_level_2_8" name="level_2_8" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_8">
                                                  Sortant de l'ASE
                                              </label>
                                              <div class="form-text" id="id_level_2_8_helptext">
                                                  Aide sociale à l'enfance
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_9_helptext" class="form-check-input" id="id_level_2_9" name="level_2_9" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_9">
                                                  DELD (12-24 mois)
                                              </label>
                                              <div class="form-text" id="id_level_2_9_helptext">
                                                  Demandeur d'emploi de longue durée (inscrit à France Travail)
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_10" name="level_2_10" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_10">
                                                  Travailleur handicapé
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_11" name="level_2_11" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_11">
                                                  Parent isolé
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_12" name="level_2_12" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_12">
                                                  Personne sans hébergement ou hébergée ou ayant un parcours de rue
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_13" name="level_2_13" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_13">
                                                  Réfugié statutaire, bénéficiaire d'une protection temporaire, protégé subsidiaire ou demandeur d'asile
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_14_helptext" class="form-check-input" id="id_level_2_14" name="level_2_14" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_14">
                                                  Résident ZRR
                                              </label>
                                              <div class="form-text" id="id_level_2_14_helptext">
                                                  Zone de revitalisation rurale
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_15_helptext" class="form-check-input" id="id_level_2_15" name="level_2_15" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_15">
                                                  Résident QPV
                                              </label>
                                              <div class="form-text" id="id_level_2_15_helptext">
                                                  Quartier prioritaire de la politique de la ville
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_16" name="level_2_16" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_16">
                                                  Sortant de détention ou personne placée sous main de justice
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_17" name="level_2_17" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_17">
                                                  Maîtrise de la langue française
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_18" name="level_2_18" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_18">
                                                  Mobilité
                                              </label>
                                          </div>
                                      </div>
                                  </fieldset>
                                  <input data-shroud-input="" name="shrouded" type="hidden" value="1"/>
                              </div>
                              <div class="row">
                                  <div class="col-12">
                                      <hr class="mb-3"/>
                                      <small class="d-inline-block mb-3">
                                          * champs obligatoires
                                      </small>
                                      <div class="form-row align-items-center justify-content-end gx-3">
                                          <div class="form-group col-12 col-lg order-3 order-lg-1">
                                              <a aria-label="Annuler la saisie de ce formulaire" class="btn btn-link btn-ico ps-lg-0 w-100 w-lg-auto" href="/job-seekers/list">
                                                  <i aria-hidden="true" class="ri-close-line ri-lg">
                                                  </i>
                                                  <span>
                                                      Annuler
                                                  </span>
                                              </a>
                                          </div>
                                          <div class="form-group col col-lg-auto order-2 order-lg-3">
                                              <button aria-label="Passer à l’étape suivante" class="btn btn-block btn-primary" type="submit">
                                                  <span>
                                                      Valider
                                                  </span>
                                              </button>
                                          </div>
                                      </div>
                                  </div>
                              </div>
                          </form>
                      </div>
                  </div>
              </div>
          </div>
      </section>
  </main>
  
  '''
# ---
# name: TestUpdateEligibilityView.test_standalone_valid_eligibility_diagnosis
  '''
  <main class="s-main" id="main" role="main">
      <div aria-atomic="true" aria-live="polite" class="toast-container">
      </div>
      <section class="s-title-02">
          <div class="s-title-02__container container">
              <div class="s-title-02__row row">
                  <div class="s-title-02__col col-12">
                      <div class="c-title">
                          <div class="c-title__main">
                              <h1>
                                  Mettre à jour
                  
                  l'éligibilité de Jane DOE
                              </h1>
                          </div>
                      </div>
                  </div>
              </div>
          </div>
      </section>
      <section class="s-section">
          <div class="s-section__container container">
              <div class="s-section__row row">
                  <div class="s-section__col col-12 col-xxl-8 col-xxxl-9">
                      <div class="c-form">
                          <div class="mb-5">
                              <p>
                                  L’éligibilité à l'IAE du candidat a été validée par :
                              </p>
                              <p class="text-tertiary fw-bold">
                                  Pres. Org.
                              </p>
                          </div>
                          <div class="alert alert-info mb-5" role="status">
                              <div class="row">
                                  <div class="col-auto pe-0">
                                      <i aria-hidden="true" class="ri-information-line ri-xl text-info">
                                      </i>
                                  </div>
                                  <div class="col">
                                      <p class="mb-2">
                                          <strong>
                                              Date de fin de validité du diagnostic : 22/09/2025
                                          </strong>
                                      </p>
                                      <p>
                                          Tant que l’éligibilité IAE est valide, vous n’avez rien à faire. Si vous souhaitez la mettre à jour, sa validité sera prolongée jusqu'au 22/09/2025.
                                      </p>
                                      <button class="btn btn-link text-start p-0" data-shroud-clear="">
                                          Mettre à jour l’éligibilité
                                      </button>
                                  </div>
                              </div>
                          </div>
                          <div id="form_errors">
                          </div>
                          <form method="post">
                              <input name="csrfmiddlewaretoken" type="hidden" value="NORMALIZED_CSRF_TOKEN"/>
                              <div class="c-info mb-3">
                                  <button class="c-info__summary">
                                      <span>
                                          Pour valider l'éligibilité IAE du candidat
                                      </span>
                                  </button>
                                  <div class="c-info__detail">
                                      <ul>
                                          <li>
                                              Veuillez vous assurer d’avoir réalisé un diagnostic socio-professionnel dans le cadre d'un entretien individuel.
                  Vous pouvez vous appuyer sur le document
                                              <a aria-label="Diagnostic socio-professionnel des candidats (ouverture dans un nouvel onglet)" class="has-external-link" href="https://communaute.inclusion.gouv.fr/surveys/dsp/create/?proconnect_login=true" rel="noreferrer noopener" target="_blank">
                                                  diagnostic socio-professionnel de référence
                                              </a>
                                              .
                                          </li>
                                          <li>
                                              Il est recommandé de sélectionner le(s) critères(s) administratifs d’éligibilité correspondants.
                                          </li>
                                      </ul>
                                  </div>
                              </div>
                              <div class="js-shroud">
                                  <fieldset>
                                      <legend class="h3">
                                          Critères administratifs de niveau 1
                                      </legend>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_1_helptext" class="form-check-input" id="id_level_1_1" name="level_1_1" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_1">
                                                  Bénéficiaire du RSA
                                              </label>
                                              <div class="form-text" id="id_level_1_1_helptext">
                                                  Revenu de solidarité active
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_2_helptext" class="form-check-input" id="id_level_1_2" name="level_1_2" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_2">
                                                  Allocataire ASS
                                              </label>
                                              <div class="form-text" id="id_level_1_2_helptext">
                                                  Allocation de solidarité spécifique
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_3_helptext" class="form-check-input" id="id_level_1_3" name="level_1_3" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_3">
                                                  Allocataire AAH
                                              </label>
                                              <div class="form-text" id="id_level_1_3_helptext">
                                                  Allocation aux adultes handicapés
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_1_4_helptext" class="form-check-input" id="id_level_1_4" name="level_1_4" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_1_4">
                                                  DETLD (+ 24 mois)
                                              </label>
                                              <div class="form-text" id="id_level_1_4_helptext">
                                                  Demandeur d'emploi de très longue durée (inscrit à France Travail)
                                              </div>
                                          </div>
                                      </div>
                                  </fieldset>
                                  <fieldset>
                                      <legend class="h3">
                                          Critères administratifs de niveau 2
                                      </legend>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_5" name="level_2_5" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_5">
                                                  Niveau d'étude 3 (CAP, BEP) ou infra
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_6" name="level_2_6" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_6">
                                                  Senior (+50 ans)
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_7" name="level_2_7" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_7">
                                                  Jeune (-26 ans)
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_8_helptext" class="form-check-input" id="id_level_2_8" name="level_2_8" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_8">
                                                  Sortant de l'ASE
                                              </label>
                                              <div class="form-text" id="id_level_2_8_helptext">
                                                  Aide sociale à l'enfance
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_9_helptext" class="form-check-input" id="id_level_2_9" name="level_2_9" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_9">
                                                  DELD (12-24 mois)
                                              </label>
                                              <div class="form-text" id="id_level_2_9_helptext">
                                                  Demandeur d'emploi de longue durée (inscrit à France Travail)
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_10" name="level_2_10" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_10">
                                                  Travailleur handicapé
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_11" name="level_2_11" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_11">
                                                  Parent isolé
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_12" name="level_2_12" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_12">
                                                  Personne sans hébergement ou hébergée ou ayant un parcours de rue
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_13" name="level_2_13" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_13">
                                                  Réfugié statutaire, bénéficiaire d'une protection temporaire, protégé subsidiaire ou demandeur d'asile
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_14_helptext" class="form-check-input" id="id_level_2_14" name="level_2_14" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_14">
                                                  Résident ZRR
                                              </label>
                                              <div class="form-text" id="id_level_2_14_helptext">
                                                  Zone de revitalisation rurale
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input aria-describedby="id_level_2_15_helptext" class="form-check-input" id="id_level_2_15" name="level_2_15" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_15">
                                                  Résident QPV
                                              </label>
                                              <div class="form-text" id="id_level_2_15_helptext">
                                                  Quartier prioritaire de la politique de la ville
                                              </div>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_16" name="level_2_16" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_16">
                                                  Sortant de détention ou personne placée sous main de justice
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_17" name="level_2_17" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_17">
                                                  Maîtrise de la langue française
                                              </label>
                                          </div>
                                      </div>
                                      <div class="form-group">
                                          <div class="form-check">
                                              <input class="form-check-input" id="id_level_2_18" name="level_2_18" type="checkbox"/>
                                              <label class="form-check-label" for="id_level_2_18">
                                                  Mobilité
                                              </label>
                                          </div>
                                      </div>
                                  </fieldset>
                                  <input data-shroud-input="" name="shrouded" type="hidden" value="1"/>
                              </div>
                              <div class="row">
                                  <div class="col-12">
                                      <hr class="mb-3"/>
                                      <small class="d-inline-block mb-3">
                                          * champs obligatoires
                                      </small>
                                      <div class="form-row align-items-center justify-content-end gx-3">
                                          <div class="form-group col-12 col-lg order-3 order-lg-1">
                                              <a aria-label="Annuler la saisie de ce formulaire" class="btn btn-link btn-ico ps-lg-0 w-100 w-lg-auto" href="/job-seekers/list">
                                                  <i aria-hidden="true" class="ri-close-line ri-lg">
                                                  </i>
                                                  <span>
                                                      Annuler
                                                  </span>
                                              </a>
                                          </div>
                                          <div class="form-group col col-lg-auto order-2 order-lg-3">
                                              <button aria-label="Passer à l’étape suivante" class="btn btn-block btn-primary" type="submit">
                                                  <span>
                                                      Valider
                                                  </span>
                                              </button>
                                          </div>
                                      </div>
                                  </div>
                              </div>
                          </form>
                      </div>
                  </div>
              </div>
          </div>
      </section>
  </main>
  
  '''
# ---
